### PR TITLE
Add speed option for test running (closes #938)

### LINKED ISSUE #938: Add `speed` option for test running
# Proposal

It's a value between 0.01 and 1. (By default it's 1). 1 - max speed

## Global speed
An option for test run:

### CLI:
 ```bash
--speed <number>
```
### [Programmatical](https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/using-testcafe/programming-interface/runner.md#run):
```js
const runner = await testCafe.createRunner();
...
runner.run({ speed: Number });
```

The options determine how fast the test will be executing. This affects to the cursor speed, delay between actions. For example you can run with more slow speed if you want to see in details what happens on your page during testing.

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/ cc @inikulin @helen-dikareva @VasilyStrelyaev

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
How does it look:
default speed     : http://screencast.com/t/WKFKnUAy
speed 0.5            : http://screencast.com/t/324IOvri
min speed (0.01) : http://screencast.com/t/zBNgRVANkPbW

File: test/server/cli-argument-parser-test.js [hunk 1]
```diff
@@ -94,6 +94,27 @@ describe('CLI argument parser', function () {
...
+
+        it('Should raise an error if the "--speed" option value is not a number', function () {
+            return assertRaisesError('--speed yo', 'Speed should be a number between 0.01 and 1.');
+        });
+
+        it('Should raise an error if the "--speed" option value less than 0.01', function () {
+            return assertRaisesError('--speed 0', 'Speed should be a number between 0.01 and 1.');
+        });
+
+        it('Should raise an error if the "--speed" option value more than 0.01', function () {
```
Comment (helen-dikareva) [inline test/server/cli-argument-parser-test.js]: `Should raise an error if the "--speed" option value more than 1`

File: src/cli/argument-parser.js [hunk 2]
```diff
@@ -68,6 +68,15 @@ export default class CLIArgumentParser {
         throw new GeneralError(MESSAGE.selectorTimeoutIsNotAnInteger);
     }
 
+    static _parseSpeed (value) {
+        value = parseFloat(value);
+
+        if (isNaN(value) || value < 0.01 || value > 1)
+            throw new GeneralError(MESSAGE.speedIsNotValidValue);
```
Comment (inikulin) [inline src/cli/argument-parser.js]: `invalidSpeedValue`

Comment (inikulin) [inline src/cli/argument-parser.js]: Also, we don't need validation here, since it will be validated in `Runner`

**Comment by AlexanderMoskovkin** [comment]
FPR
(one test for speed changing in iframe added)

**Comment by inikulin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
ping @VasilyStrelyaev

File: src/cli/argument-parser.js [hunk 3]
```diff
@@ -145,6 +145,7 @@ export default class CLIArgumentParser {
             .option('-f, --fixture <name>', 'run only fixtures with the specified name')
             .option('-F, --fixture-grep <pattern>', 'run only fixtures matching the specified pattern')
             .option('--selector-timeout <ms>', 'set the amount of time within which selectors make attempts to obtain a node to be returned')
+            .option('--speed <number>', 'set the speed of tests execution (0.01 ... 1)')
```
Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]:
```
            .option('--speed <factor>', 'set the speed of test execution (0.01 ... 1)')
```

**Comment by AlexanderMoskovkin** [comment]
FPR

**Pull request merged**